lib/matplotlib/backend_bases.py

```diff
@@ -1649,9 +1663,13 @@ def _fix_ipython_backend2gui(cls):
             # Work around pylabtools.find_gui_and_backend always reading from
             # rcParamsOrig.
             orig_origbackend = mpl.rcParamsOrig["backend"]
+
             try:
                 mpl.rcParamsOrig["backend"] = mpl.rcParams["backend"]
-                ip.enable_matplotlib()
+                if is_non_interactive_terminal_ipython(ip):
```

Comment:
Can you 1) make the function private (leading underscore) and 2) perhaps move the check a bit earlier, e.g. in line 1649 `if not ip or _is_non_interactive...(ip)`?